clang/test/CodeGenSYCL/local_var_big_init_as.cpp

```diff
@@ -0,0 +1,34 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice \
+// RUN: -emit-llvm -o - %s | FileCheck %s
+
+// This test checks that data for big constant initializer lists is placed
+// into the global address space by the SYCL compiler.
+
+struct Test {
+  Test() : set{
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15,
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15
+    // CHECK-DAG: @constinit = private unnamed_addr addrspace(1) constant
+    // CHECK: [32 x i32]
+    // CHECK: [i32 0, i32 1, i32 2, i32 3, i32 4, i32 5, i32 6, i32 7,
+    // CHECK:  i32 8, i32 9, i32 10, i32 11, i32 12, i32 13, i32 14, i32 15,
+    // CHECK:  i32 0, i32 1, i32 2, i32 3, i32 4, i32 5, i32 6, i32 7,
+    // CHECK:  i32 8, i32 9, i32 10, i32 11, i32 12, i32 13, i32 14, i32 15
+    // CHECK  ], align 4
+    // CHECK-NOT: @constinit = private unnamed_addr addrspace(0)
+    // CHECK-NOT: @constinit = private unnamed_addr addrspace(2)
+    // CHECK-NOT: @constinit = private unnamed_addr addrspace(3)
+    // CHECK-NOT: @constinit = private unnamed_addr addrspace(4)
+  } {};
+  int set[32];
+};
+
+__attribute__((sycl_device)) void bar(Test &x);
+
+__attribute__((sycl_device)) void zoo(int *y) {
```

Comment:
Can you please explain the need for `zoo `and `vec `in this test?

Reply:
`zoo` contains the definition of a vector with static initializer, bar uses it in such a manner that compiler can't replace usage with individual constants. This makes sure `@constinit` global is generated.

Reply:
I'm not sure if `vec` array in `zoo()` makes a difference to `set ` initializer. I don't know if I am missing something but I don't see how it is being used in this test.  I think you can delete the array in `zoo ` and the parameter ` int *y`. I was curious to see the IR without these and so I checked out your patch and ran the test without it. There is no difference in IR. The `@constinit` global is still generated

Reply:
Right, `vec` and `y` are redundant. My reply was just about `zoo` basically. Will fix now.